### PR TITLE
Improve parser accuracy

### PR DESCRIPTION
**Pull request by Andrewed95**

### CODE REVIEW

File: play.py [hunk 1]
```diff
@@ -164,14 +160,11 @@
...
         pinv2 = Api.api.get_doc("Purchase Invoice", pinv2.name)
-        validate(
-            instance=pinv1,
-            schema=JSON1_DATA_SCHEMA,
-        )
-        pr['json1'] = json.dumps(pinv1)
-        pr['json2'] = json.dumps(pinv2)
-        pr['doctype'] = 'PreRechnung'
-        Api.api.update(pr)
+        if validate_json1(pinv1):
```
Comment (tillmo) [inline play.py]: There should be a function for just validating all PreRechnungs, without modifying anything,

File: purchase_invoice.py [hunk 2]
```diff
@@ -176,8 +176,7 @@ def search_item(self, supplier, check_dup=True):
         texts = ['Neuen Artikel anlegen']
         texts += [i[1]['item_code'] + ' ' + i[1]['item_name'] for i in top_items]
         title = "Artikel wählen"
-        msg = "Artikel in Rechnung:\n{0}\n\nCode Lieferant: {1}\n\n".format(self.long_description,self.item_code)
-              
+        msg = "Artikel in Rechnung:\n{0}\n\nCode Lieferant: {1}\n\n".format(self.long_description, self.item_code)
```
Comment (tillmo) [inline purchase_invoice.py:179]: all changes in this file seem to concern reformatting. Therefore, this file should not be part of the pull request.

**Review by tillmo — CHANGES_REQUESTED** [review]

File: purchase_invoice.py (hunk 2 above)
Comment (Andrewed95) [reply purchase_invoice.py:179]: No, there are some important changes:
1. line 943 and line 945
2. line 952 and line 953

File: play.py (hunk 1 above)
Comment (Andrewed95) [reply play.py]: This block was removed.

Comment (tillmo) [reply play.py]: There should be a function for validating all PreRechnungs.

Comment (Andrewed95) [reply play.py]: Added
